plugins/task-plugin/src/machine/websocket.ts

```diff
@@ -28,22 +31,102 @@ export class ReconnectingWebSocket {
 
     private readonly logger: Logger;
 
-    constructor(url: string) {
-        this.url = url;
+    constructor(targetUrl: string) {
+        this.url = targetUrl;
         this.logger = new ConsoleLogger();
         this.open();
     }
 
     private getOptions(): WS.ClientOptions {
         let options: WS.ClientOptions = {};
-        if (fs.existsSync(SS_CRT_PATH)) {
-            const SS_SRT = fs.readFileSync(SS_CRT_PATH).toString();
-            options = { ca: SS_SRT };
+
+        const certificateAuthority = this.getCertificateAuthority();
+
+        const proxyUrl = process.env.http_proxy;
+        const baseUrl = process.env.CHE_API;
+        if (proxyUrl && proxyUrl !== '' && baseUrl) {
+            const parsedBaseUrl = url.parse(baseUrl);
+            if (parsedBaseUrl.hostname && this.shouldProxy(parsedBaseUrl.hostname)) {
+                const parsedProxyUrl = url.parse(proxyUrl);
+                const mainProxyOptions = this.getMainProxyOptions(parsedProxyUrl);
+                const httpsProxyOptions = this.getHttpsProxyOptions(mainProxyOptions, parsedBaseUrl.hostname, certificateAuthority);
+                const httpOverHttpAgent = tunnel.httpOverHttp({ proxy: mainProxyOptions });
+                const httpOverHttpsAgent = tunnel.httpOverHttps({ proxy: httpsProxyOptions });
+                const httpsOverHttpAgent = tunnel.httpsOverHttp({
+                    proxy: mainProxyOptions,
+                    ca: certificateAuthority ? [certificateAuthority] : undefined
+                });
+                const httpsOverHttpsAgent = tunnel.httpsOverHttps({
+                    proxy: httpsProxyOptions,
+                    ca: certificateAuthority ? [certificateAuthority] : undefined
+                });
+                const urlIsHttps = (parsedBaseUrl.protocol || 'http:').startsWith('https:');
```

Comment:
parsedUrl.protocol && parsedUrl.startsWith('https') would be easier to understand, IMO.